sympy/geometry/point.py

```diff
@@ -266,8 +266,14 @@ def distance(self, p):
         sqrt(x**2 + y**2)
 
         """
-        p = Point(p)
-        return sqrt(sum([(a - b)**2 for a, b in zip(self.args, p.args)]))
+        try:
+            args = p.args
+        except AttributeError:
+            try:
+                args = [as_int(i) for i in p]
+            except TypeError:
```

Comment:
I am skeptical about this way, I rather like the instance checking  method as suggested here: https://github.com/sympy/sympy/pull/10455#issuecomment-174329614

Since, this one takes the same time ~ 70ns & complicates the things unnecessarily.